src/ucp/core/ucp_worker.c

```diff
@@ -385,6 +477,118 @@ void ucp_worker_progress(ucp_worker_h worker)
     ucs_assert(--worker->inprogress == 0);
 }
 
+ucs_status_t ucp_worker_get_efd(ucp_worker_h worker, int *fd)
+{
+    int res, res_fd;
+    ucs_status_t status;
+    ucp_rsc_index_t tl_id;
+    struct epoll_event event;
+    ucp_context_h context = worker->context;
+
+    if (worker->wakeup.wakeup_efd != -1) {
+        *fd = worker->wakeup.wakeup_efd;
+        return UCS_OK;
+    }
+
+    res_fd = epoll_create(context->num_tls);
+    if (res_fd == -1) {
+        ucs_error("Failed to create epoll descriptor");
+        return UCS_ERR_IO_ERROR;
+    }
+
+    event.events = EPOLLIN | EPOLLET;
+    event.data.fd = worker->wakeup.wakeup_pipe[0];
+    res = epoll_ctl(res_fd, EPOLL_CTL_ADD, worker->wakeup.wakeup_pipe[0], &event);
+    if (res == -1) {
+        ucs_error("Failed to add pipe descriptor to epoll");
+        return UCS_ERR_IO_ERROR;
+    }
+
+    for (tl_id = 0; tl_id < context->num_tls; tl_id++) {
+        if (worker->wakeup.iface_wakeups[tl_id] != NULL) {
+            status = ucp_worker_wakeup_attach(res_fd,
+                    worker->wakeup.iface_wakeups[tl_id]);
+            if (status != UCS_OK) {
+                goto epoll_cleanup;
+            }
+        }
+    }
+
+    worker->wakeup.wakeup_efd = res_fd;
+    *fd = res_fd;
+    return UCS_OK;
+
+epoll_cleanup:
+    close(res_fd);
+    return status;
+}
+
+ucs_status_t ucp_worker_arm(ucp_worker_h worker)
+{
+    int res;
+    char buf;
+    ucs_status_t status;
+    ucp_rsc_index_t tl_id;
+    uct_wakeup_h wakeup_iter;
```

Comment:
why call it "wakeup_iter" - your iterator is "tl_id". just call it "wakeup" or "uct_wakeup"...